### ISSUE TITLE
web3modal ether5 connect error

### ISSUE THREAD

**Issue by CjoyHome** [safe-global/safe-wallet-web#3037]
## Bug description
An error occurs when I use ether v5 in web3modal v3 to connect to safe
![image](https://github.com/safe-global/safe-wallet-web/assets/26813427/60ab0c9f-cf9f-4b1b-9739-b84694ce00d0)
![image](https://github.com/safe-global/safe-wallet-web/assets/26813427/a661e3d7-2e7b-450d-a251-ba511a8ff552)

It is mentioned in this issue that the safe side needs help to solve the problem. I hope I can get help.
https://github.com/WalletConnect/web3modal/issues/1574

## Environment
 - Browser: Chrome
 - Wallet: MetaMask
 - Chain: Ethereum mainnet

NPM package
```
"@walletconnect/modal": "^2.6.2",
"@web3modal/ethers5": "^3.4.0",
"ethers": "^5.7.2",
"next": "12.1.6", 
"react": "18.1.0", 
"react-bootstrap": "^2.4.0", 
"react-dom": "18.1.0", 
"react-intl": "^6.4.2"
....
```

**Comment by katspaugh** [comment]
We indeed don't support chainId 0, as the error message say.

**Comment by glitch-txs** [comment]
Hi @katspaugh, thanks for taking a look, this is going to be fixed on our end in the next release https://github.com/WalletConnect/web3modal/pull/1553
However I wanted to let you know that this configuration exists in the optional namespaces, so Safe should still work/connect with this configuration.

**Comment by katspaugh** [comment]
We only throw that error when a request comes on the wrong chain (`event.params.chainId` in the `session_request` event). It should still connnect fine if an unsupported chainId is in the optional namespace.

**Comment by CjoyHome** [comment]
How should I solve this problem?

**Comment by CjoyHome** [comment]
> How should I solve this problem?

@glitch-txs 

Currently I am using version 3.4.0, how should I deal with this problem?

**Comment by glitch-txs** [comment]
The fix was merged in 3.5.1
